smack-extensions/src/main/java/org/jivesoftware/smackx/message_correct/MessageCorrectExtension.java

```diff
@@ -83,11 +81,13 @@ public String getElementName() {
      * @see org.jivesoftware.smack.packet.PacketExtension#toXML()
      */
     @Override
-    public CharSequence toXML() {
-        final StringBuilder stringBuilder = new StringBuilder();
-        stringBuilder.append('<').append(getElementName()).append(' ').append(JID_TAG).append('=').append('\'')
-                .append(getJidInitialMessage()).append('\'').append(" xmlns='").append(getNamespace()).append("'/>");
-        return stringBuilder.toString();
+    public XmlStringBuilder toXML() {
+        XmlStringBuilder xml = new XmlStringBuilder(this, getNamespace());
+        xml.attribute(ID_TAG, getIdInitialMessage());
+        xml.attribute("xmlns", getNamespace());
```

Comment:
> But I want to show the id attribute before the xmlns attribute.

Why? The order is completely irrelevant, but it's common practice to have the unprefixed `xmlns` attribute first (if one is required). This is why the XmlStringBuilder API is designed this way.